packages/runner/src/context.ts

```diff
@@ -108,26 +113,31 @@ export function createTestContext(
   context.onTestFailed = (handler, timeout) => {
     test.onFailed ||= []
     test.onFailed.push(
-      withTimeout(handler, timeout ?? runner.config.hookTimeout, true),
+      withTimeout(handler, timeout ?? runner.config.hookTimeout, true, new Error('STACK_TRACE_ERROR')),
     )
   }
 
   context.onTestFinished = (handler, timeout) => {
     test.onFinished ||= []
     test.onFinished.push(
-      withTimeout(handler, timeout ?? runner.config.hookTimeout, true),
+      withTimeout(handler, timeout ?? runner.config.hookTimeout, true, new Error('STACK_TRACE_ERROR')),
     )
   }
 
   return runner.extendTaskContext?.(context) || context
 }
 
-function makeTimeoutMsg(isHook: boolean, timeout: number) {
-  return `${
+function makeTimeoutError(isHook: boolean, timeout: number, stackTraceError?: Error) {
+  const message = `${
     isHook ? 'Hook' : 'Test'
   } timed out in ${timeout}ms.\nIf this is a long-running ${
     isHook ? 'hook' : 'test'
   }, pass a timeout value as the last argument or configure it globally with "${
     isHook ? 'hookTimeout' : 'testTimeout'
   }".`
+  const error = new Error(message)
```

Comment:
wouldn't it be more performant to change the message and stack trace of the original error?

Reply:
Since we don't read this new error's `error.stack`, `prepareStackTrace` rewriting only applies to 
 `stackTraceError.stack`, so this won't matter for the performance.

I realized that we reuse a single `stackTraceError.stack` for potential multiple hook timeout errors from the same `beforeEach`, so mutating the original `stackTraceError` isn't probably good.

If we go extreme, we can return a plain object from runner like `{ message, stack }`, but I don't think that's necessary.